libr/anal/fcn.c

```diff
@@ -1516,9 +1516,13 @@ R_API int r_anal_fcn_cc(RAnal *anal, RAnalFunction *fcn) {
 
 R_API char *r_anal_function_get_signature(RAnalFunction *function) {
 	RAnal *a = function->anal;
+	char *fcnname = strdup (function->name);
+	if (!strncmp (function->name, "sym.imp.", 8)) {
+		fcnname = strdup (function->name + 8);
```

Comment:
fcnname is leaking here

Reply:
imho we should differentitiate between fcnname and signname.. not sure how we should manage this, maybe holding the original signature info...